pkg/routeagent_driver/handlers/healthchecker/healthchecker_test.go

```diff
@@ -0,0 +1,277 @@
+
+					t.pingerMap[healthCheckIP1].AwaitStart()
+					t.pingerMap[healthCheckIP2] = fake.NewPinger(healthCheckIP2)
+				})
+
+				It("should replace the pinger if the HealthCheckIP changes", func() {
+					endpoint1.Spec.HealthCheckIP = healthCheckIP2
+					err := t.handler.RemoteEndpointUpdated(endpoint1)
+					Expect(err).NotTo(HaveOccurred())
+
+					t.pingerMap[healthCheckIP1].AwaitStop()
+					t.pingerMap[healthCheckIP2].AwaitStart()
+				})
+			})
+		})
+		When("a RemoteEndpoint is deleted", func() {
+			var endpoint1 *submarinerv1.Endpoint
+
+			BeforeEach(func() {
+				endpoint1 = t.CreateEndpoint(t.newSubmEndpoint())
+				err := t.handler.RemoteEndpointCreated(endpoint1)
+				Expect(err).NotTo(HaveOccurred())
+
+				t.pingerMap[healthCheckIP1].AwaitStart()
+			})
+
+			It("should stop and remove the pinger for the deleted HealthCheckIP", func() {
+				err := t.handler.RemoteEndpointRemoved(endpoint1)
+				Expect(err).NotTo(HaveOccurred())
+
+				t.pingerMap[healthCheckIP1].AwaitStop()
+			})
+		})
+		When("transitioning to a non-gateway state", func() {
+			BeforeEach(func() {
+				endpoint1 := t.CreateEndpoint(t.newSubmEndpoint())
+				err := t.handler.RemoteEndpointCreated(endpoint1)
+				Expect(err).NotTo(HaveOccurred())
+				t.pingerMap[healthCheckIP1].AwaitStart()
+			})
+
+			It("should start pingers for all remote endpoints", func() {
+				err := t.handler.TransitionToGateway()
+				Expect(err).NotTo(HaveOccurred())
+
+				t.pingerMap[healthCheckIP1].AwaitStop()
+
+				err = t.handler.TransitionToNonGateway()
+				Expect(err).NotTo(HaveOccurred())
+
+				t.pingerMap[healthCheckIP1].AwaitStart()
+			})
+		})
+	})
+})
+
+type testDriver struct {
+	*eventtesting.ControllerSupport
+	pingerMap     map[string]*fake.Pinger
+	handler       event.Handler
+	endpoints     dynamic.ResourceInterface
+	client        submarinerv1client.RouteAgentInterface
+	stopCh        chan struct{}
+	localNodeName string
+}
+
+func newTestDriver() *testDriver {
+	t := &testDriver{
+		ControllerSupport: eventtesting.NewControllerSupport(),
+	}
+
+	BeforeEach(func() {
+		t.stopCh = make(chan struct{})
+		scheme := runtime.NewScheme()
+		Expect(submarinerv1.AddToScheme(scheme)).To(Succeed())
+		Expect(submarinerv1.AddToScheme(kubeScheme.Scheme)).To(Succeed())
+
+		clientset := fakeClient.NewSimpleClientset()
+
+		dynamicClient := dynamicfake.NewSimpleDynamicClient(kubeScheme.Scheme)
+
+		t.endpoints = dynamicClient.Resource(submarinerv1.SchemeGroupVersion.WithResource("endpoints")).Namespace(namespace)
+		t.client = clientset.SubmarinerV1().RouteAgents(namespace)
+		t.pingerMap = map[string]*fake.Pinger{
+			healthCheckIP1: fake.NewPinger(healthCheckIP1),
+		}
+
+		t.localNodeName = "nodeName"
+
+		config := &healthchecker.Config{
+			PingInterval:             1, // Set interval to 1 second for faster testing
+			MaxPacketLossCount:       1,
+			HealthCheckerEnabled:     true,
+			RouteAgentUpdateInterval: 2 * time.Second,
```

Comment:
You can make this a lot less to speed up the tests

```suggestion
			RouteAgentUpdateInterval: 100 * time.Millisecond,
```

Reply:
Done.